### PR TITLE
NotificationsMenu

### PR DESCRIPTION
**Pull request by flynn1999**
WIP: This commit contains the menu with style along with functioning time stamp from hardcoded data.

### CODE REVIEW

File: stories/notifications/Notifications.stories.js [hunk 1]
```diff
@@ -0,0 +1,1162 @@
...
+// const logger =
+//     this.props.logger ||
+//     ((Notification) => {
+//         console.log(Notification);
+//     });
+
+return (
+    <NotificationsPreview
+        // presenter={presenter}
+        baseDebugId="notificationWindow"
```
Comment (ash3rz) [inline stories/notifications/Notifications.stories.js:1153]: Looks like you have all that notification mock data, but you're not using it. It's possible you didn't have any data to work with until after you had already begun putting stuff together. I would really recommend that your dev process begins with using mock data and iterating your components over that data.  This will save you a TON of time in the long-run.

**Comment by flynn1999** [comment]
I'll begin changing everything to whatever already exists in material ui. It's great that this was reviewed now so that I can fix all this before proceeding with anything else.

File: stories/notifications/Notifications.stories.js (hunk 1 above)
Comment (flynn1999) [reply stories/notifications/Notifications.stories.js:1153]: Yes, I had this added recently. I did not think there would be too much of a difference with replacing hard coded data with mock data in the end. In what ways does it help to save time?

File: src/pages/NotificationsMenu.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    return dayDiff == 0 && (
+        diff < 60 && "Just Now" ||
+        diff < 120 && "A minute ago" ||
+        diff < 3600 && Math.floor( diff / 60 ) + " minutes ago" ||
+        diff < 7200 && "An hour ago" ||
+        diff < 86400 && Math.floor( diff / 3600 ) + " hours ago") ||
+        dayDiff == 1 && "Yesterday" ||
+        dayDiff < 7 && dayDiff + " days ago" ||
+        dayDiff > 31 && Math.ceil(dayDiff / 7) + " weeks ago";
```
[the remark this answers is not in the document]
Comment (flynn1999) [reply src/pages/NotificationsMenu.js:52]: As far as I could find date-fns does not have one single function to help with making the date simple but it has functions to subtract days and time. I'll be replacing the calculations with those functions but the main "framework" of the function will remain the same.

[the remark this answers is not in the document]
Comment (ianmcorvidae) [reply src/pages/NotificationsMenu.js:52]: so, date-fns definitely does have this! In fact, it has several variations depending on needs: [`formatDistance`](https://date-fns.org/v2.15.0/docs/formatDistance), [`formatDistanceStrict`](https://date-fns.org/v2.15.0/docs/formatDistanceStrict), and [`formatRelative`](https://date-fns.org/v2.15.0/docs/formatRelative), plus the two `Distance` ones also have `DistanceToNow` variants that automatically pull in "now" so you don't have to pass that in manually -- these functions appear in two different places I could find -- in the `i18n` section, but also in the "common helpers" below that in the documentation sidebar. One thing to note which may or may not be relevant to you, is that elsewhere we're using date-fns' `lightFormat` rather than `format` -- this was a size tradeoff; lightFormat doesn't support every token and I think doesn't support locales. The `Distance` and `Relative` functions don't appear to have equivalents that differ in the same ways, which may mean that they'll pull in the non-`light` version of the regular formatter too, and if they do we may want to switch the other date formatting to use the full-fledged `format` too. That would probably fall outside of what you'd be expected to do, but it's the sort of thing I like to mention at times like this.

File: stories/notifications/Notifications.stories.js (hunk 1 above)
Comment (ash3rz) [reply stories/notifications/Notifications.stories.js:1153]: Just to be clear, hard-coded data is not the issue. Mock data is also hard-coded data after all :smile: The issue is how you're using that data. From a high level, it's causing you to add an extra step.

Right now, you're baking the data directly into the component, making it a static component, in order to get a feel for the look of things and get feedback. Once that's complete, then you'll need to remove the baked in data and also refactor your static component into a dynamic component that can work with real/mocked data.  It's like you're creating the same component twice. Storybook is really great for eliminating that extra step because it allows you to bake your data into a story instead so your component shouldn't have to be refactored from static to dynamic.

Beyond that, knowing from the start what the data looks like and working with it directly can save time. Maybe you've crafted your component around certain assumptions about the data and then come to find out the data is different than you expected. For example, the conversation about notification length came up in Slack. Some notifications are longer than maybe you originally thought, so then we started the conversation about using ellipses or not. Just as an example.

File: src/pages/NotificationsMenu.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (flynn1999) [reply src/pages/NotificationsMenu.js:52]: I totally skipped that. Thank you for informing. I was searching for something along the lines of "Subtract time" and clearly did not browse through the docs well enough. I did not completely understand what you meant by 
> "The Distance and Relative functions don't appear to have equivalents that differ in the same ways, which may mean that they'll pull in the non-light version of the regular formatter too". 
What problems would we face exactly?

[the remark this answers is not in the document]
Comment (ianmcorvidae) [reply src/pages/NotificationsMenu.js:52]: The `lightFormat` function is different in only one way that matters to us, basically, which is that it's smaller as far as transfer size. If we have need of another function that happens to bring in the full version of `format` (which, it's worth noting, I don't know for sure that these do! I'm just guessing they might), then there's no benefit of us using `lightFormat` any more, and in fact might be slightly to our detriment since we'd be using both versions.

Basically, it's a small performance benefit to not send as much javascript to our users, so it's better, one way or another, to only be using one of the two. But it won't cause any immediate bugs.
